### ISSUE TITLE
setBacklight() not compatible with LCD API 1.0

### ISSUE THREAD

**Issue by bperrybap** [adafruit/Adafruit_LiquidCrystal#6]
The setBacklight() function in the adafruit_LiquidCrystal library is not compatible with the way other libraries have implemented this function.
Other libraries that have implemented this function use the argument to this function as a dimvalue rather than an absolute status as specified in the LCD API 1.0
http://playground.arduino.cc/Code/LCDAPI

This allows dimming on h/w that supports it.
For h/w that does not support dimming, any non zero value will turn the backlight on.

Unfortunately, the way the code is implemented in the Adafruit_LiquidCrystal library only a value of HIGH will turn on the backlight and all other values will turn off the backlight.
This breaks sketches that have been written to be portable across multiple libraries and transparently offer dimming on h/w that supports it.

The Adafruit_LiquidCrystal library can easily be updated to correct/fix this by simply checking for a non zero value and use that to turn on the backlight rather than accepting any 8 bit value but only turning on the backlight  for a value of HIGH.

This could be corrected in a number of ways but the simplest is probably to just make a tweak to setBacklight()

setBacklight() can be corrected to be:

```
// Allows to set the backlight, if the LCD backpack is used
void Adafruit_LiquidCrystal::setBacklight(uint8_t val) {
  if(val) // no dimming so any non zero value turns on backlight
        val = HIGH;
  // check if i2c or SPI
  if ((_i2cAddr != 255) || (_SPIclock != 255)) {
    _digitalWrite(7, val); // backlight is on pin 7
  }
}
```

This will update should not affect any existing Adafruit_LiquidCrystal sketches, and will update the library code to be compatible with the LCD API 1.0 so the library will no longer break existing portable sketches written for multiple libraries that now are also supporting the Adafruit_LiquidCrystal library.

On a related note, I would also recommend considering changing the code in 
Adafruit_LiquidCrystal::_digitalWrite() and Adafruit_MCP23008::digitatlWrite()
as those functions currently explicitly check for HIGH and if not HIGH then set the pin output to low.
Given that the functions do not use a boolean argument value , and can be handed 8 bit values other than LOW or HIGH, I think it would be wiser to check for LOW, and set the pin high if the value is not LOW.
This would seem to be a better way of handling the argument value given it is not a boolean.

**Comment by bperrybap** [comment]
Also, one of the advantages of using the argument as a dimvlaue rather than an absolute pin state is that application does not have to be aware of the polarity of the backllight control output pin. 
i.e. some devices use an active low signal to turn on a backlight. This can all be hidden when the API avoids using absolute pin state information and lets the low level h/w code deal with it by abstracting the values passed to mean backlight states rather than pin states.

Also, on a related not, while not in the LCD API 1.0 spec, many libraries have implemented 
backlight() and noBacklight() functions to turn on on/off the backlight.
These functions also avoid promoting knowledge of backlight control pin state to the application and also provide the added benefit that the application does not have to know the maximum brightness value needed to set the backlight to its maximum brightness.

On h/w that do not support dimming, these functions set the backlight to full off or full on.

**Comment by caternuson** [comment]
Should be fixed by #16.
